hedera-node/src/main/java/com/hedera/services/state/exports/StandardSigFileWriter.java

```diff
@@ -15,25 +15,42 @@
  */
 package com.hedera.services.state.exports;
 
+import static com.hedera.services.exports.FileCompressionUtils.COMPRESSION_ALGORITHM_EXTENSION;
+
 import com.google.common.primitives.Ints;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.UncheckedIOException;
+import java.util.zip.GZIPOutputStream;
 
 public class StandardSigFileWriter implements SigFileWriter {
+
+    static final String SIG_FILE_EXTENSION = "_sig";
+
     @Override
-    public String writeSigFile(String signedFile, byte[] sig, byte[] signedFileHash) {
-        var sigFile = signedFile + "_sig";
-        try (FileOutputStream fout = new FileOutputStream(sigFile, false)) {
-            fout.write(TYPE_FILE_HASH);
-            fout.write(signedFileHash);
-            fout.write(TYPE_SIGNATURE);
-            fout.write(Ints.toByteArray(sig.length));
-            fout.write(sig);
+    public String writeSigFile(
+            final String signedFilePath, final byte[] sig, final byte[] signedFileHash) {
+        final var isCompressed = signedFilePath.endsWith(COMPRESSION_ALGORITHM_EXTENSION);
```

Comment:
Will this also compress record file signature files? If so, we need to fix it so that only account balance signatures are compressed.

Reply:
The signature file creation for record files is done [entirely in RecordStreamFileWriter](https://github.com/hashgraph/hedera-services/blob/8a6e44df337aeb803593ce89edee345e823596ca/hedera-node/src/main/java/com/hedera/services/stream/RecordStreamFileWriter.java#L618-L644).

Reply:
Thanks for the clarification